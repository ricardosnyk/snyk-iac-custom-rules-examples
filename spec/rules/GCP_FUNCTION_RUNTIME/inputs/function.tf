provider "google" {
  project     = "my-project-id"
  region      = "us-central1"
}

resource "google_cloudfunctions_function" "valid" {
  name        = "function-test"
  description = "My function"
  runtime     = "nodejs16"

  available_memory_mb   = 128
  source_archive_bucket = google_storage_bucket.bucket.name
  source_archive_object = google_storage_bucket_object.archive.name
  trigger_http          = true
  entry_point           = "helloGET"
}

resource "google_cloudfunctions_function" "invalid" {
  name        = "function-test"
  description = "My function"
  runtime     = "nodejs8"

  available_memory_mb   = 128
  source_archive_bucket = google_storage_bucket.bucket.name
  source_archive_object = google_storage_bucket_object.archive.name
  trigger_http          = true
  entry_point           = "helloGET"
}

resource "google_cloudfunctions_function" "invalid2" {
  name        = "function-test"
  description = "My function"
  runtime     = "go111"

  available_memory_mb   = 128
  source_archive_bucket = google_storage_bucket.bucket.name
  source_archive_object = google_storage_bucket_object.archive.name
  trigger_http          = true
  entry_point           = "helloGET"
}

resource "google_cloudfunctions_function" "valid2" {
  name        = "function-test"
  description = "My function"
  runtime     = "python38"

  available_memory_mb   = 128
  source_archive_bucket = google_storage_bucket.bucket.name
  source_archive_object = google_storage_bucket_object.archive.name
  trigger_http          = true
  entry_point           = "helloGET"
}