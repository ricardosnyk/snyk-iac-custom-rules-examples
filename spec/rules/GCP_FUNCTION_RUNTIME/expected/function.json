[
  {
    "passed": false,
    "ignored": false,
    "resource_id": "google_cloudfunctions_function.invalid",
    "resource_namespace": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
    "resource_type": "google_cloudfunctions_function",
    "severity": "high",
    "resources": [
      {
        "id": "google_cloudfunctions_function.invalid",
        "type": "google_cloudfunctions_function",
        "namespace": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
        "location": [
          {
            "filepath": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
            "line": 18,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "runtime"
            ],
            "location": {
              "filepath": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
              "line": 21,
              "column": 3
            }
          }
        ]
      }
    ]
  },
  {
    "passed": false,
    "ignored": false,
    "resource_id": "google_cloudfunctions_function.invalid2",
    "resource_namespace": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
    "resource_type": "google_cloudfunctions_function",
    "severity": "high",
    "resources": [
      {
        "id": "google_cloudfunctions_function.invalid2",
        "type": "google_cloudfunctions_function",
        "namespace": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
        "location": [
          {
            "filepath": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
            "line": 30,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "runtime"
            ],
            "location": {
              "filepath": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
              "line": 33,
              "column": 3
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "google_cloudfunctions_function.valid",
    "resource_namespace": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
    "resource_type": "google_cloudfunctions_function",
    "severity": "high",
    "resources": [
      {
        "id": "google_cloudfunctions_function.valid",
        "type": "google_cloudfunctions_function",
        "namespace": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
        "location": [
          {
            "filepath": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
            "line": 6,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "runtime"
            ],
            "location": {
              "filepath": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
              "line": 9,
              "column": 3
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "google_cloudfunctions_function.valid2",
    "resource_namespace": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
    "resource_type": "google_cloudfunctions_function",
    "severity": "high",
    "resources": [
      {
        "id": "google_cloudfunctions_function.valid2",
        "type": "google_cloudfunctions_function",
        "namespace": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
        "location": [
          {
            "filepath": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
            "line": 42,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "runtime"
            ],
            "location": {
              "filepath": "spec/rules/GCP_FUNCTION_RUNTIME/inputs/function.tf",
              "line": 45,
              "column": 3
            }
          }
        ]
      }
    ]
  }
]